en/ros/ros2_offboard_control.md

```diff
@@ -0,0 +1,190 @@
+        publish_offboard_control_mode();
+        publish_position_setpoint_triplet();
+
+        // stop the counter after reaching 100
+        if (offboard_setpoint_counter_ < 101) {
+                offboard_setpoint_counter_++;
+        }
+};
+timer_ = this->create_wall_timer(10ms, timer_callback);
+```
+
+The above is the main loop spining on the ROS 2 node.
+It first sends 100 setpoint messages before sending the command to change to offboard mode
+At the same time, both `offboard_control_mode` and `position_setpoint_triplet` messages are sent to the flight controller.
+
+```c++
+/**
+ * @brief Publish the offboard control mode.
+ *        For this example, only position and altitude controls are active.
+ */
+void OffboardControl::publish_offboard_control_mode() const {
+	OffboardControlMode msg{};
+	msg.timestamp = timestamp_.load();
+	msg.ignore_thrust = true;
+	msg.ignore_attitude = true;
+	msg.ignore_bodyrate_x = true;
+	msg.ignore_bodyrate_y = true;
+	msg.ignore_bodyrate_z = true;
+	msg.ignore_position = false;
+	msg.ignore_velocity = true;
+	msg.ignore_acceleration_force = true;
+	msg.ignore_alt_hold = true;
+
+	offboard_control_mode_publisher_->publish(msg);
+}
+
+/**
+ * @brief Publish position setpoint triplets.
+ *        For this example, it sends position setpoint triplets to make the
+ *        vehicle hover at 5 meters.
+ */
+void OffboardControl::publish_position_setpoint_triplet() const {
+	PositionSetpointTriplet msg{};
+	msg.timestamp = timestamp_.load();
+	msg.current.timestamp = timestamp_.load();
+	msg.current.type = PositionSetpoint::SETPOINT_TYPE_POSITION;
+	msg.current.x = 0.0;
+	msg.current.y = 0.0;
+	msg.current.z = -5.0;
+	msg.current.cruising_speed = -1.0;
+	msg.current.position_valid = true;
+	msg.current.alt_valid = true;
+	msg.current.valid = true;
+
+	position_setpoint_triplet_publisher_->publish(msg);
+}
+```
+
+The above functions exemplify how the fields on both `offboard_control_mode` and `position_setpoint_triplet` messages can be set.
+Notice that the above example is applicable for offboard position control, where on the `offboard_control_mode` message, the `ignore_position` field is set to `true`, while all the others get set to `false`, and in the `position_setpoint_triplet`, on the `current` setpoint, `valid`, `alt_valid` and `position_valid` are set to `true`.
+Also, in this case, `x`, `y` and `z` fields are hardcoded to certain values, but they can be updated dynamically according to an algorithm or even by a subscription callback for messages coming from another node.
+
+:::tip
+The position is already being published in the NED coordinate frame for simplicity, but in the case of the user wanting to subscribe to data coming from other nodes, and since the standard frame of reference in ROS/ROS2 is ENU, the user can use the available helper functions in the [`frame_transform` library](https://github.com/PX4/px4_ros_com/blob/master/src/lib/frame_transforms.cpp).
```

Comment:
This is a little confusing. 

I think you are saying "here we have a note that is already converted to NED". If you are using a "normal" ROS2 node the data will be in ENU and you will need to convert it. In that case you can use this library. Is that correct?

Reply:
Yes that is correct. Can you suggest a way of rewording it?